pallets/allocations/src/benchmarking.rs

```diff
@@ -22,27 +22,37 @@
 
 use super::*;
 use crate::BalanceOf;
-#[cfg(test)]
 use crate::Pallet as Allocations;
 use frame_benchmarking::{account, benchmarks};
-use frame_support::{traits::ConstU32, BoundedVec};
+use frame_support::{
+	traits::{ConstU32, Hooks},
+	BoundedVec,
+};
 use frame_system::RawOrigin;
 use sp_std::prelude::*;
 
 pub type MaxMembers = ConstU32<10>;
 
 const SEED: u32 = 0;
+const ALLOC_FACTOR: u32 = 10;
 
 fn make_batch<T: Config>(b: u32) -> BoundedVec<(T::AccountId, BalanceOf<T>), T::MaxAllocs> {
 	let mut ret = BoundedVec::with_bounded_capacity(b as usize);
 
 	for i in 0..b {
 		let account = account("grantee", i, SEED);
-		let _ = ret.try_push((account, T::ExistentialDeposit::get() * 10u32.into()));
+		let _ = ret.try_push((account, T::ExistentialDeposit::get() * ALLOC_FACTOR.into()));
 	}
 	ret
 }
 
+fn assert_last_event<T: Config>(generic_event: <T as Config>::Event) {
+	let events = frame_system::Pallet::<T>::events();
+	let system_event: <T as frame_system::Config>::Event = generic_event.into();
+	let frame_system::EventRecord { event, .. } = &events[events.len() - 1];
```

Comment:
It can and it would lead to an assertion which is actually the required behaviour of this function.  Also this function is taken from the other parity pallets and used exactly the same way in upstream.

Reply:
if you add assert(events.len()=>1 ) it will be clearer this it the intended behavior

Reply:
I added an assert for further clarification.